flink-connector-vitess-cdc/src/test/java/com/vervetica/cdc/connectors/vitess/VitessSourceTest.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.vervetica.cdc.connectors.vitess;
+
+import org.apache.flink.api.common.state.BroadcastState;
+import org.apache.flink.api.common.state.KeyedStateStore;
+import org.apache.flink.api.common.state.ListState;
+import org.apache.flink.api.common.state.ListStateDescriptor;
+import org.apache.flink.api.common.state.MapStateDescriptor;
+import org.apache.flink.api.common.state.OperatorStateStore;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.testutils.CheckedThread;
+import org.apache.flink.runtime.state.FunctionInitializationContext;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.util.MockStreamingRuntimeContext;
+import org.apache.flink.util.Collector;
+
+import com.ververica.cdc.connectors.utils.TestSourceContext;
+import com.ververica.cdc.connectors.vitess.VitessSource;
+import com.ververica.cdc.connectors.vitess.config.TabletType;
+import com.ververica.cdc.connectors.vitess.config.VtctldConfig;
+import com.ververica.cdc.debezium.DebeziumDeserializationSchema;
+import com.ververica.cdc.debezium.DebeziumSourceFunction;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.Statement;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import static com.ververica.cdc.connectors.utils.AssertUtils.assertDelete;
+import static com.ververica.cdc.connectors.utils.AssertUtils.assertInsert;
+import static com.ververica.cdc.connectors.utils.AssertUtils.assertUpdate;
+
+/** Tests for {@link VitessSource} which also heavily tests {@link DebeziumSourceFunction}. */
+public class VitessSourceTest extends VitessTestBase {
+
+    @Before
+    public void before() {
+        initializeMysqlTable("inventory");
+    }
+
+    @Test
+    public void testConsumingAllEvents() throws Exception {
```

Comment:
I find the VitessConnectorITCase also test consume all events? I think we can just keep one of them.

Reply:
VitessSourceTest tests DataStream API's SourceFunction and VitessConnectorITCase tests Table/SQL API
, so I think both of them are needed.